mypy/dmypy_server.py

```diff
@@ -270,11 +271,17 @@ def cmd_stop(self) -> Dict[str, object]:
     def cmd_run(self, version: str, args: Sequence[str]) -> Dict[str, object]:
         """Check a list of files, triggering a restart if needed."""
         try:
-            sources, options = mypy.main.process_options(
-                ['-i'] + list(args),
-                require_targets=True,
-                server_options=True,
-                fscache=self.fscache)
+            # Process options can exit on improper arguments, so we need to catch that and
+            # capture stderr so the client can report it
+            stderr = io.StringIO()
+            with redirect_stderr(stderr):
+                sources, options = mypy.main.process_options(
+                    ['-i'] + list(args),
+                    require_targets=True,
+                    server_options=True,
+                    fscache=self.fscache,
+                    program='dmypy',
```

Comment:
I'm not sure I agree with this -- it's technically not `dmypy` that cannot parse the argument. Maybe set this to `mypy-daemon` so it's clear that the error comes from the daemon process?

Reply:
Yeah, good idea. I wanted to somehow signal it wasn't mypy complaining about this.